docs/design/descheduler.md

```diff
@@ -0,0 +1,115 @@
+* Node status changes such as add/remove nodes, pod/node taint/affinity changes.
+
+In order to rebalance the cluster resource utilization among nodes, introduce **DeScheduler** to achieve these goals:
+* Rescheduling pods based on real resource utilization instead of request resource.
+* Support custom configured rescheduling strategies.
+
+## Design
+### Architecture
+![Volcano work with Descheduler](./images/descheduler-architecture.jpg)
+1. Metrics are collected in `node_exporter` and uploaded to `Prometheus` for all nodes. The metrics contains basic metrics
+(cpu/memory) and self-defined metrics.
+2. `Volcano` binds pods to fit nodes. `kubelet` will be charge of the lifecycle of pods.
+3. `DeScheduler` lists all nodes and pods, filters the pods not match the rescheduling strategies and deletes them. All 
+the decisions are made by real node resource utilization provided by `Prometheus`.
+4. Pod deleted by `Descheduler` will be recreated by ReplicaSet and rescheduled by `Volcano`. 
+5. Execute the rescheduling process above periodically.
+
+### Rescheduling Strategy
+* OfflineOnly
+
+    `OfflineOnly`, abbreviated as `OLO`, will only pick out offline workloads, which are attached with annotation
+`preemptable: true`, and then reschedule them only.
+
+* LowPriorityFirst
+
+    `LowPriorityFirst`, abbreviated as `LPF`, will sort workloads by priority and then reschedule pods in ascending 
+order.
+
+* ShortLifeTimeFirst
+
+  `ShortLifeTimeFirst`, abbreviated as `SLTF`, will sort workloads by running time. Pods with the shortest lift time 
+will be rescheduled first. This strategy can make sure workloads with long task type goes healthily without interruptions.
+
+* BigJobFirst
+
+    `BigJobFirst`, abbreviated as `BJF`, will select jobs which request the most **dominate resource** and reschedule
+them first. It helps improve system throughout and avoid small jobs' starvation.
+
+* MoreReplicasFirst
+  
+    `MoreReplicasFirst`, abbreviated as `MRF`, will sort workloads by replicas number. Workloads with the most replicas
+will be rescheduled first. This strategy is friendly to `gang scheduling` for it will consider the `minAvailable` in 
+volcano jobs.
+
+### Job Filter
+* Queue Filter
+
+    `Queue filtering` will filter jobs in specified queue. Then the rescheduling process only works on the result set.
+
+* Label Filter
+
+    `Label Filter` will filter jobs with specified labels. Then the rescheduling process only works on the result set.
+
+### Metrics
+All the decisions made by `DeScheduler` will consider the metrics from `Prometheus`. Namely, `DeScheduler` will list the
+real node resource utilization and pod distribution instead of request resource. Basically, usage of`CPU` and `Memroy` 
+will be collected. Other resource such as `GPU` can be extended.
+
+## Implementation Plan
+1. Fork from [kubernetes-sigs/descheduler](https://github.com/kubernetes-sigs/descheduler) to implement the basic ability.
```

Comment:
Please do not fork it; it's hard to maintain another repo :(

Reply:
I'm seeking a way to implement custom strategies without forking the upstream repository. Waiting for the answer from https://github.com/kubernetes-sigs/descheduler/issues/686